django/db/models/expressions.py

```diff
@@ -1620,6 +1620,20 @@ def get_source_expressions(self):
     def set_source_expressions(self, exprs):
         self.condition, self.result = exprs
 
+    def resolve_expression(
+        self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False
+    ):
+        """
+        Ensure condition is always resolved with for_save=False since it's used
+        as a filter, not as an update value.
+        """
```

Comment:
I think the comment valuable but I'm not sure it's worth _obscuring_ the inherited `__doc__` string with it. Any thoughts about using `#` instead of a string literal so it doesn't get picked up?

```suggestion
        # Ensure condition is always resolved with for_save=False since it's used
        # as a filter, not as an update value.
```